### PR TITLE
Add flake8 pre-commit hook. Change black hook to also format notebooks.

### PR DESCRIPTION
**Pull request by nbiederbeck**
I like having my code checked by pyflakes/flake8, but I dislike getting told so by our CI.

Personally, I think it is better to have this checked locally, to prevent "fix pyflakes"-commits.

I chose flake8, because it wraps pyflakes among others

--

`black-jupyter` formats python files *and* notebooks

### CODE REVIEW

**Comment by kosack** [comment]
This means we'll have to fix all the remaining code style warnings (since pycodestyle is run by flake8), but probably that is a good thing!  I just checked and there are not so many (that aren't already disabled in `setup.cfg`)

**Comment by nbiederbeck** [comment]
Yes, but only in files one commits

**Comment by maxnoe** [comment]
> but I dislike getting told so by our CI.

Pyflakes checks for actual errors, not style issues (with the exception of unused imports). This is an important first check in the CI and if that fails, there is no point in running the unit tests in the CI.

Of course it would be best if everyone ran all the checks locally first, but that does not make running it in the CI any less necessary.

**Comment by nbiederbeck** [comment]
**RE pre-commit/CI**: I like having a short/fast feedback loop. Having the CI fail with pyflakes annoys me, and pyflakes is really fast on a local machine (as opposed to the full test-suite). I'm not for replacing, I'm for additional checks that makes the life of developers easier (at least "easier imho").

**RE pyflakes/flake8**: according to pyflakes maintainers (as seen in a [PR I opened](https://github.com/PyCQA/pyflakes/pull/709#issuecomment-1156383725)) the preferred way of using pyflakes as pre-commit hook is via flake8. flake8 is a wrapper around pyflakes and pycodestyle, so we get more than "just" pyflakes.

**RE isort**: I'm in

**RE black compability**: The [black docs](https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html) give configurations for other tools, which I added in my last commit.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
     hooks:
       - id: black-jupyter
+  # https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html?highlight=other%20tools#flake8
   - repo: https://github.com/PyCQA/flake8
     rev: 4.0.1
     hooks:
       - id: flake8
+        args: [--max-line-length=88, --extend-ignore=E203]
+  # https://pycqa.github.io/isort/docs/configuration/black_compatibility.html#integration-with-pre-commit
+  - repo: https://github.com/pycqa/isort
```
Comment (maxnoe) [inline .pre-commit-config.yaml:13]: Are these run in order? Then it might make sense to run isort then Black, then flake8.

**Review by maxnoe — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**